pkgs/swift2objc/lib/src/parser/parsers/declaration_parsers/parse_struct_declaration.dart

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import '../../../ast/declarations/compounds/members/method_declaration.dart';
+import '../../../ast/declarations/compounds/members/property_declaration.dart';
+import '../../../ast/declarations/compounds/struct_declaration.dart';
+import '../../_core/json.dart';
+import '../../_core/parsed_symbolgraph.dart';
+import '../../_core/utils.dart';
+import '../parse_declarations.dart';
+
+StructDeclaration parseStructDeclaration(
```

Comment:
This looks very similar to `parseClassDeclaration`. What if you have `parseStructDeclaration` and `parseClassDeclaration` both be thin wrappers around a `parseCompoundDeclaration` function? I think the only thing you'd need to change is to have `parseCompoundDeclaration` take a constructor tear-off: `parseCompoundDeclaration(classSymbolJson, symbolgraph, StructDeclaration.new)`.

The type of the constructor tear-off `Function` would be a bit complicated since it takes named params. I think it's something like `CompoundDeclaration Function({id: String, name: String, methods: List<MethodDeclaration>, properties: List<PropertyDeclaration>})`.

Reply:
Didn't know about tear-offs. Very neat syntax. I pretty much refactored the whole 2 functions into one.